plugins/twitter/classes/DataProvider/Twitter.php

```diff
@@ -82,24 +84,40 @@ public function fetch($limit = FALSE) {
 				"result_type" => 'recent'
 			]);
 
-			if ( empty($results->statuses))
+			if ( empty($results['statuses']))
 			{
 				return 0;
 			}
 
-			$statuses = $results->statuses;
+			$statuses = $results['statuses'];
 
 			// Store the highest id
-			$this->since_id = $statuses[0]->id;
+			$this->since_id = $statuses[0]['id'];
 
 			foreach ($statuses as $status) {
-				$id = $status->id;
-				$user = $status->user;
-				$screen_name = $user->screen_name;
-				$text = $status->text;
+				$id = $status['id'];
+				$user = $status['user'];
+				$screen_name = $user['screen_name'];
+				$text = $status['text'];
+
+				$additional_data = [];
+
+				if ($status['coordinates'] || $status['place']) {
+					$additional_data['location'] = [];
+					if ($status['coordinates']) {
+						$additional_data['location'][] = $status['coordinates'];
+					}
+
+					if ($status['place'] && $status['place']['bounding_box']) {
+						// Find center of bounding box.
```

Comment:
Considering add https://github.com/brick/geo as it would replace our existing gisconverter libs too